### PR TITLE
[MISC] Split edit distance matrix tests into multiple ones

### PR DESCRIPTION
**Pull request by marehr**
`test/unit/alignment/matrix/edit_distance_score_matrix_full_test.cpp` and `test/unit/alignment/matrix/edit_distance_trace_matrix_full_test.cpp` became kind of big. The main issue is that they test different matrix representations in different modes and it is hard to keep track which mode tests what. This change extracts all test that belong to the same mode and puts them into own tests.

This is part of #1568

### CODE REVIEW

File: test/unit/alignment/matrix/edit_distance_matrix/edit_distance_score_matrix.hpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+std::vector<std::vector<int>> as_row_wise_vector(auto matrix)
+{
+    std::vector<std::vector<int>> result{};
+    for (unsigned row = 0; row < matrix.rows(); ++row)
+    {
+        result.push_back({});
+        for (unsigned col = 0; col < matrix.cols(); ++col)
+        {
+            std::optional<int> entry = matrix.at({detail::row_index_type{row}, detail::column_index_type{col}});
```
Comment (joergi-w) [inline test/unit/alignment/matrix/edit_distance_matrix/edit_distance_score_matrix.hpp]:
```suggestion
            std::optional<score_type> entry = matrix.at({detail::row_index_type{row}, detail::column_index_type{col}});
```

File: test/unit/alignment/matrix/edit_distance_matrix/edit_distance_score_matrix.hpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    {}
+
+    using base_t::max_rows;
+    using base_t::add_column;
+    using base_t::reserve;
+};
+
+static constexpr score_type INF = detail::matrix_inf<score_type>;
+
+std::vector<std::vector<int>> as_row_wise_vector(auto matrix)
```
Comment (joergi-w) [inline test/unit/alignment/matrix/edit_distance_matrix/edit_distance_score_matrix.hpp]:
```suggestion
std::vector<std::vector<score_type>> as_row_wise_vector(auto matrix)
```

File: test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_global_max_errors_test.cpp [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+#include "edit_distance_score_matrix.hpp"
+
+TEST(global_max_errors, empty)
+{
+    matrix_type<false, true> matrix{1u};
+
+    // row-wise matrix
+    std::vector<std::vector<int>> result = as_row_wise_vector(matrix);
+    std::vector<std::vector<int>> expect{{}};
```
Comment (joergi-w) [inline test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_global_max_errors_test.cpp]: use score_type (and also change other occurrences in all new files of this PR)

File: test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_global_max_errors_test.cpp [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+TEST(global_max_errors, epsilon)
+{
+    matrix_type<false, true> matrix{1u};
+
+    matrix.add_column({}, {}, 1u);
+
+    // row-wise matrix
+    std::vector<std::vector<int>> result = as_row_wise_vector(matrix);
+    std::vector<std::vector<int>> expect{{-0}};
```
Comment (joergi-w) [inline test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_global_max_errors_test.cpp]: `-0`?

File: test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_global_max_errors_test.cpp [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+    matrix.add_column({}, {}, 1u);
+    matrix.add_column({}, {}, 1u);
+    matrix.add_column({}, {}, 1u);
+    matrix.add_column({}, {}, 0u);
+    matrix.add_column({}, {}, 0u);
+
+    // row-wise matrix
+    std::vector<std::vector<int>> result = as_row_wise_vector(matrix);
+    std::vector<std::vector<int>> expect{{-0, -1, -2,INF,INF}};
```
Comment (joergi-w) [inline test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_global_max_errors_test.cpp]: Space after comma:
```suggestion
    std::vector<std::vector<int>> expect{{-0, -1, -2, INF, INF}};
```

File: test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_semi_global_max_errors_test.cpp [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+    matrix.add_column({}, {}, 1u);
+    matrix.add_column({}, {}, 1u);
+    matrix.add_column({}, {}, 1u);
+    matrix.add_column({}, {}, 1u);
+    matrix.add_column({}, {}, 1u);
+
+    // row-wise matrix
+    std::vector<std::vector<int>> result = as_row_wise_vector(matrix);
+    std::vector<std::vector<int>> expect{{-0, -0, -0, -0, -0}};
```
Comment (joergi-w) [inline test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_semi_global_max_errors_test.cpp]: `-0` again

File: test/unit/alignment/matrix/edit_distance_matrix/trace_matrix_unbanded_global_max_errors_test.cpp [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+    {
+        {N  ,l  ,l  ,l  ,l  ,l  ,l  ,l  ,N  ,N  },
+        {u  ,D  ,Dl ,l  ,l  ,l  ,l  ,l  ,N  ,N  },
+        {u  ,u  ,D  ,D  ,Dl ,l  ,l  ,l  ,N  ,N  },
+        {u  ,u  ,Du ,Du ,D  ,D  ,Dl ,l  ,N  ,N  },
+        {N  ,u  ,Du ,Du ,Du ,Du ,D  ,D  ,N  ,N  },
+        {N  ,N  ,D  ,Dul,N  ,N  ,N  ,N  ,N  ,N  },
+        {N  ,N  ,N  ,D  ,N  ,N  ,N  ,N  ,N  ,N  },
+        {N  ,N  ,N  ,N  ,N  ,N  ,N  ,N  ,N  ,N  },
+        {N  ,N  ,N  ,N  ,N  ,N  ,N  ,N  ,N  ,N  }
```
Comment (joergi-w) [inline test/unit/alignment/matrix/edit_distance_matrix/trace_matrix_unbanded_global_max_errors_test.cpp]: Why are D and N capital letters and the others not?

File: test/unit/alignment/matrix/edit_distance_matrix/edit_distance_score_matrix.hpp [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+    matrix_type(size_t const rows_size) : base_t{rows_size}
+    {}
+
+    using base_t::max_rows;
+    using base_t::add_column;
+    using base_t::reserve;
+};
+
+static constexpr score_type INF = detail::matrix_inf<score_type>;
```
[the remark this answers is not in the document]
Comment (marehr) [reply test/unit/alignment/matrix/edit_distance_matrix/edit_distance_score_matrix.hpp]: like we talked about, there will be a followup which will fix this.

**Comment by marehr** [comment]
@joergi-w  https://github.com/seqan/seqan3/projects/10#card-32366237

File: test/unit/alignment/matrix/edit_distance_matrix/edit_distance_trace_matrix.hpp [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+#include "../../pairwise/fixture/alignment_fixture.hpp"
+
+using namespace seqan3;
+using namespace seqan3::test::alignment::fixture;
+
+using word_type = uint8_t;
+
+template <bool is_semi_global, bool use_max_errors>
+class matrix_type
+    : public seqan3::detail::edit_distance_trace_matrix_full<word_type, is_semi_global, use_max_errors>
```
Comment (rrahn) [inline test/unit/alignment/matrix/edit_distance_matrix/edit_distance_trace_matrix.hpp]: 💅 please put the colon on the previous line.

File: test/unit/alignment/matrix/edit_distance_matrix/edit_distance_trace_matrix.hpp [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <seqan3/alignment/matrix/edit_distance_trace_matrix_full.hpp>
+#include <seqan3/alignment/matrix/row_wise_matrix.hpp>
+
+#include "../../pairwise/fixture/alignment_fixture.hpp"
+
+using namespace seqan3;
```
Comment (rrahn) [inline test/unit/alignment/matrix/edit_distance_matrix/edit_distance_trace_matrix.hpp]: You are going to remove the global namespaces in the future PR.

File: test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_global_max_errors_test.cpp [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+
+    EXPECT_EQ(result, expect);
+}
+
+TEST(global_max_errors, single_word_1)
```
Comment (rrahn) [inline test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_global_max_errors_test.cpp]: single_word_1?
The number has a meaning, or it is just an enumeration of different test cases?

File: test/unit/alignment/matrix/edit_distance_matrix/edit_distance_trace_matrix.hpp (hunk 9 above)
Comment (marehr) [reply test/unit/alignment/matrix/edit_distance_matrix/edit_distance_trace_matrix.hpp]: I will fix that in the followup PR.

File: test/unit/alignment/matrix/edit_distance_matrix/edit_distance_trace_matrix.hpp (hunk 10 above)
Comment (marehr) [reply test/unit/alignment/matrix/edit_distance_matrix/edit_distance_trace_matrix.hpp]: I'll do it in the follow up

File: test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_global_max_errors_test.cpp (hunk 11 above)
Comment (marehr) [reply test/unit/alignment/matrix/edit_distance_matrix/score_matrix_unbanded_global_max_errors_test.cpp]: I fixed that in the follwup.

**Comment by rrahn** [comment]
@marehr you can now rebase if you want. Otherwise I can squash commit it

**Pull request merged**